src/Entity.php

```diff
@@ -414,9 +426,76 @@ public function prepareInputForUpdate($input)
             $input = $this->checkRightDatas($input);
         }
 
+        $input = $this->handleCustomScenesInputValues($input);
+
+        return $input;
+    }
+
+    private function handleCustomScenesInputValues(array $input): array
+    {
+        $fields = [
+            "custom_helpdesk_home_scene_left",
+            "custom_helpdesk_home_scene_right",
+        ];
+
+        foreach ($fields as $field) {
+            // Check for special values
+            $value = $input[$field] ?? "";
+            if (empty($value)) {
+                continue;
+            }
+
+            if ($value == self::SCENE_DEFAULT) {
+                // Reset default value (empty string)
+                $input[$field] = "";
+            } elseif ($value == self::SCENE_INHERIT) {
+                // Inherit parent value
+                $input[$field] = self::CONFIG_PARENT;
+            } elseif ($value == self::SCENE_CUSTOM) {
+                // A custom file was submitted
+                $files = $input["_$field"] ?? [];
+                if (!is_array($files) || empty($files)) {
+                    // Unexpected format or no files were submitted, do not
+                    // modify the saved value.
+                    unset($input[$field]);
+                } else {
+                    $file = array_pop($files);
+                    $input[$field] = $this->handleCustomScenesSubmittedFile($file);
+                }
+            }
+        }
+
         return $input;
     }
 
+    private function handleCustomScenesSubmittedFile(string $file): string
+    {
+        // Read file path
+        $path = realpath(GLPI_TMP_DIR . "/$file");
+        if (!$path || !str_starts_with($path, GLPI_TMP_DIR)) {
+            // File doest not exist or is outside upload directory
+            $message = __("An unexpected error occured.");
+            Session::addMessageAfterRedirect($message);
+            return "";
```

Comment:
IMHO, we should not process the update in this case. Indeed, if a valid value was previously defined, you will erase it with an empty value.
This method may return `false`, and this `false` value could be propagated by `handleCustomScenesInputValues`.